bluemira/equilibria/opt_constraint_funcs.py

```diff
@@ -95,9 +95,9 @@ def Ax_b_constraint(constraint, vector, grad, a_mat, b_vec, value, scale):  # no
     scale: float
         Current scale with which to calculate the constraints
     """
-    constraint[:] = np.dot(a_mat, scale * vector) - b_vec - value
+    constraint[:] = a_mat @ vector * scale - b_vec - value
     if grad.size > 0:
-        grad[:] = scale * a_mat
+        grad[:] = scale * a_mat.T
```

Comment:
Neet to test the optimization whith a gradient based method to see if this is a valid correction

Reply:
Agreed - I think it's worth adding a test to check this constraint (and future ones) for gradient based methods (eg. SLSQP) explicitly in `test_constraints`, given how easy it would be for a newly written constraint to get the shape of grad wrong! 

FWIW - on a closer look, I don't think the transpose should be on L100 - I think it's already included in `a_mat` from L560 in `opt_constraints.py`. Easy to confirm with a quick test, though!

Reply:
Reverted to initial value

Reply:
After test on the STEP equilibrium designeer the transpose should be not there.
As for the the reason the test does not catch it I do not understand it.